### PR TITLE
feat: integrate new way to get partition candidates for hot compaction

### PR DESCRIPTION
**Pull request by NGA-TRAN**
Another PR for https://github.com/influxdata/idpe/issues/16836

This is for hot compaction only. Most changes in this PR is to remove tests of  across-all-shards because we no longer need to consider shards; and at appropriate creation time for test partition to work with our new way

Note that: My next PR will do for COLD partitions where I need to add extra code to consider whether to compact them or not

- [ ] I've read the contributing section of the project [README](https://github.com/influxdata/influxdb_iox/blob/main/README.md).
- [ ] Signed [CLA](https://influxdata.com/community/cla/) (if not already signed).

### CODE REVIEW

File: compactor/src/hot.rs [hunk 1]
```diff
@@ -181,37 +139,26 @@ pub(crate) async fn hot_partitions_to_compact(
...
         let partitions = repos
-            .parquet_files()
-            .recent_highest_throughput_partitions(
-                shard_id,
-                hours_ago_in_ns,
-                min_number_recent_ingested_files_per_partition,
-                max_number_partitions_per_shard,
-            )
+            .partitions()
+            .partitions_with_recent_created_files(hours_ago_in_ns)
```
Comment (NGA-TRAN) [inline compactor/src/hot.rs:150]: This is the main change to read partition table instead

File: compactor/src/hot.rs [hunk 2]
```diff
@@ -92,56 +89,17 @@ pub(crate) async fn hot_partitions_to_compact(
...
-            let mut partitions = hot_partitions_for_shard(
-                Arc::clone(&compactor.catalog),
-                None,
-                &query_times,
-                min_number_recent_ingested_files_per_partition,
-                max_number_partitions_per_shard,
-            )
-            .await?;
+    let mut partitions =
+        hot_partitions_for_shard(Arc::clone(&compactor.catalog), &query_times).await?;
```
Comment (NGA-TRAN) [inline compactor/src/hot.rs:93]: With the new way to get candidates, we no longer need to consider shard anymore

Comment (pauldix) [inline compactor/src/hot.rs:93]: rename this function given it's no longer by shard?

**Review by pauldix — APPROVED** [review]
One small comment to rename a func, but otherwise LGTM

**Pull request merged**